pkg/connector/backfill.go

```diff
@@ -19,11 +19,19 @@ import (
 
 // getTakeoutID blocks until the takeout ID is available.
 func (t *TelegramClient) getTakeoutID(ctx context.Context) (takeoutID int64, err error) {
-	log := zerolog.Ctx(ctx).With().Str("function", "getTakeoutID").Logger()
+	// Always reset the timer
+	defer func() {
+		if t.stopTakeoutTimer != nil {
+			t.stopTakeoutTimer.Stop()
+		}
```

Comment:
This stop could probably happen already before the loop (immediately after checking that there's no existing takeout)

Reply:
does this look right? https://github.com/mautrix/telegramgo/pull/33/commits/c0c2ab972865c9e6a61b6e8c8a47dba4671d5d0a